src/github.com/getlantern/flashlight/flashlight.go

```diff
@@ -83,7 +85,53 @@ func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
+func logPanic(proxyAddr string, msg string) {
+	cfg, err := config.Init(packageVersion)
+	if err != nil {
+		panic("Error initializing config")
+	}
+
+	// Create the client-side proxy.
+	c := &client.Client{
+		Addr:         proxyAddr,
+		ReadTimeout:  0, // don't timeout
+		WriteTimeout: 0,
+	}
+	ready := make(chan bool)
+	go c.ListenAndServe(func() { ready <- true })
+	<-ready
+
+	if err := logging.Init(); err != nil {
+		panic("Error initializing logging")
+	}
+
+	ready = logging.Configure(proxyAddr, cfg.CloudConfigCA, cfg.InstanceId, version, revisionDate)
+	<-ready
+
+	log.Error(msg)
+
+	logging.Flush()
+	_ = logging.Close()
+}
+
 func main() {
+	// panicwrap works by re-executing the running program (retaining arguments,
+	// environmental variables, etc.) and monitoring the stderr of the program.
+	exitStatus, err := panicwrap.BasicWrap(
+		func(output string) {
+			logPanic("127.0.0.1:18787", output)
+			exit(nil)
+		})
+	if err != nil {
+		// Something went wrong setting up the panic wrapper. This wont' be
+		// captured by panicwrap
+		log.Fatalf("Error setting up panic wrapper: %v", err)
+	}
+	// If exitStatus >= 0, then we're the parent process.
+	if exitStatus >= 0 {
+		os.Exit(exitStatus)
```

Comment:
I'm similarly not sure we should just exit here @uaalto. If we're the parent process, maybe we just log the fact and keep going? What do you think?